assets/js/admin-settings/rule.jsx

```diff
@@ -0,0 +1,145 @@
+/* global wp, wpLocalizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import { useSelect } from '@wordpress/data';
+import { ruleCard } from './styles';
+
+
+const Rule = ({
+  field,
+  loading,
+  newRule,
+  onAdd,
+  onDelete,
+  onUpdate,
```

Comment:
`onUpdate` should handle the index in the parent. It should accept a field and a value (e.g., `onUpdate('taxonomy', 'category')`.